pkg/tfgen/generate.go

```diff
@@ -1483,18 +1486,19 @@ func input(sch shim.Schema, info *tfbridge.SchemaInfo) bool {
 //
 //	would need to understand how to unmarshal names in a language-idiomatic way (and specifically reverse the
 //	name transformation process).  This isn't impossible, but certainly complicates matters.
-func propertyName(key string, sch shim.Schema, custom *tfbridge.SchemaInfo) string {
+func propertyName(key string, sch shim.SchemaMap, custom map[string]*tfbridge.SchemaInfo) string {
 	// Use the name override, if one exists, or use the standard name mangling otherwise.
 	if custom != nil {
-		if custom.Name != "" {
-			return custom.Name
+		v := custom[key]
```

Comment:
Oof. Is this blocks till needed? The TerraformToPulumiNameV2 bit takes care of SchemaInfo.Name now, no? Or is the worry about the `// BUGBUG` effects ?

Reply:
That block can be yanked. The `// BUGBUG` only effects the key, which is not used in `.Name` override lookup.